File: php/features.php
<?php
declare(strict_types=1);

// RemoteJobs SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class RemoteJobsFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new RemoteJobsBaseFeature();
            case "test":
                return new RemoteJobsTestFeature();
            default:
                return new RemoteJobsBaseFeature();
        }
    }
}

File: py/features.py
# RemoteJobs SDK feature factory

from feature.base_feature import RemoteJobsBaseFeature
from feature.test_feature import RemoteJobsTestFeature


def _make_feature(name):
    features = {
        "base": lambda: RemoteJobsBaseFeature(),
        "test": lambda: RemoteJobsTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
